staging/src/k8s.io/client-go/tools/clientcmd/validation.go

```diff
@@ -235,10 +234,15 @@ func validateAuthInfo(authInfoName string, authInfo clientcmdapi.AuthInfo) []err
 				validationErrors = append(validationErrors, fmt.Errorf("unable to read client-key %v for %v due to %v", authInfo.ClientKey, authInfoName, err))
 			}
 		}
+		methods = append(methods, "tlsClientAuth")
```

Comment:
TLS client auth in combination with bearer token auth is valid and intentionally allowed